src/parsers/manifest/dash/parse_mpd.ts

```diff
@@ -277,5 +275,34 @@ function parseCompleteIntermediateRepresentation(
     }
   }
 
+  let duration: number|undefined;
+
+  const { minimumTime, maximumTime } = parsedMPD;
+  if (maximumTime?.isContinuous === false &&
+      minimumTime?.isContinuous === false) {
+    duration = (maximumTime.value - minimumTime.value);
+  } else {
+    if (rootAttributes.duration !== undefined) {
+      const firstPosition = parsedPeriods.reduce((acc: number|undefined, { start }) => {
```

Comment:
In a MPD, Periods are in chronological order, no? There's no point for that reduce.

Reply:
Right